### PR TITLE
Add socket.io with multiple cluster's support

### PR DESCRIPTION
**Pull request by bertho-zero**
For activate socket.io compatibility in production mode, or with multiple clusters, set clusterSticky to true in your config file.

For complete my pull request on meanio: https://github.com/linnovate/meanio/pull/93

### CODE REVIEW

**Comment by timelf123** [comment]
Need to review whether or not this uses the incoming IP address for load balancing purposes. If it does ignore for-forwarding headers, then in production behind a load balancer as you want to run on multiple machines for redundancy purposes or just scale out, all node processes will receive the IP address of load balancer causing all requests to go to a single process.

**Comment by bertho-zero** [comment]
I using HAProxy for load balancing and it is no problem

> we must add the `cookie SERVERID insert indirect nocache` statement. This tells HAProxy to add a new parameter called SERVERID to our user’s cookie. This line is integral for ensuring that our user “sticks” to the same socket.io process - which inturn maintains our user’s state.

File: server.js [hunk 1]
```diff
@@ -19,7 +20,8 @@ var deferred = require('q').defer();
 if ((cluster.isMaster) &&
   (process.execArgv.indexOf('--debug') < 0) &&
   (process.env.NODE_ENV!=='test') && (process.env.NODE_ENV!=='development') &&
-  (process.execArgv.indexOf('--singleProcess')<0)) {
+  (process.execArgv.indexOf('--singleProcess')<0) &&
+  (!config.clusterSticky)) {
```
Comment (ADSKLowenthal) [inline server.js]: !config.clusterSticky here seems to contradict with https://github.com/linnovate/meanio/pull/93/files#diff-d508c9002324dfcd2364791530ab44d7R127

**Comment by bertho-zero** [comment]
No, look below I thought everyone normally. https://github.com/linnovate/meanio/pull/93/files#diff-d508c9002324dfcd2364791530ab44d7R129

**Comment by timelf123** [comment]
This combined with https://github.com/linnovate/meanio/pull/93 worked for me, thanks!

**Pull request closed without merging**